lib/ValidationError.spec.ts

```diff
@@ -194,6 +194,152 @@ describe('fromZodError()', () => {
       }
     }
   });
+
+  test('handles nested zod.object() schema errors', () => {
+    const objSchema = zod.object({
+      id: zod.number().int().positive(),
+      arr: zod.array(zod.number().int()),
+      nestedObj: zod.object({
+        name: zod.string().min(2),
+      }),
+    });
+
+    try {
+      objSchema.parse({
+        id: -1,
+        arr: [1, 'a'],
+        nestedObj: {
+          name: 'a',
+        },
+      });
+    } catch (err) {
+      if (err instanceof ZodError) {
+        const validationError = fromZodError(err);
+        expect(validationError).toBeInstanceOf(ValidationError);
+        expect(validationError.message).toMatchInlineSnapshot(
+          `"Validation error: Number must be greater than 0 at "id"; Expected number, received string at "arr[1]"; String must contain at least 2 character(s) at "nestedObj.name""`
+        );
+        expect(validationError.details).toMatchInlineSnapshot(`
+                  [
+                    {
+                      "code": "too_small",
+                      "exact": false,
+                      "inclusive": false,
+                      "message": "Number must be greater than 0",
+                      "minimum": 0,
+                      "path": [
+                        "id",
+                      ],
+                      "type": "number",
+                    },
+                    {
+                      "code": "invalid_type",
+                      "expected": "number",
+                      "message": "Expected number, received string",
+                      "path": [
+                        "arr",
+                        1,
+                      ],
+                      "received": "string",
+                    },
+                    {
+                      "code": "too_small",
+                      "exact": false,
+                      "inclusive": true,
+                      "message": "String must contain at least 2 character(s)",
+                      "minimum": 2,
+                      "path": [
+                        "nestedObj",
+                        "name",
+                      ],
+                      "type": "string",
+                    },
+                  ]
+              `);
+      }
+    }
+  });
+
+  test('handles zod.or() schema errors', () => {
+    const success = zod.object({
+      status: zod.literal('success'),
+      data: zod.object({
+        id: zod.string(),
+      }),
+    });
+
+    const error = zod.object({
+      status: zod.literal('error'),
+    });
+
+    const objSchema = success.or(error);
+
+    try {
+      objSchema.parse(
+        {},
+        {
+          path: ['responce-schema'],
+        }
```

Comment:
thought: unsure whether we should be passing `ParseParams` as part of this test. It seems to be a separate test case.

Reply:
moved to separate test